### PR TITLE
UIEH-693: Create note page

### PR DESCRIPTION
**Pull request by MaksymDryha**
This PR adds a create note page. It makes use of `NoteCreatePage` component introduced recently in stripes-smart-components.

### CODE REVIEW

File: src/routes/note-create.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { Redirect } from 'react-router';
+
+import { NoteCreatePage } from '@folio/stripes-smart-components';
```
Comment (Yurii-Danylenko) [inline src/routes/note-create.js]: Please, change path to `@folio/stripes/smart-components`

File: src/routes/note-create.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      goBack: PropTypes.func.isRequired,
+    }).isRequired,
+    location: ReactRouterPropTypes.location.isRequired,
+  };
+
+  getReferredEntityData() {
+    const {
+      entityName: name,
+      entityType: type,
+      entityId: id,
```
Comment (Godlevskyi) [inline src/routes/note-create.js]: Please update the readme for the connectet component to reflect the change from the queryparams to location state

Comment (MaksymDryha) [reply src/routes/note-create.js]: @Godlevskyi, the connected component receives referred entity data as a prop. Query params or location.state are used in apps' routes

File: test/bigtest/tests/note-create-test.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      });
+    });
+
+    describe('when the note create page is visited using a link', () => {
+      beforeEach(function () {
+        this.visit('/eholdings/notes/new');
+      });
+
+      it('should redirect to eholdings root route', function () {
+        expect(this.location.pathname + this.location.search).to.equal('/eholdings?searchType=providers');
```
Comment (carolegodfrey) [inline test/bigtest/tests/note-create-test.js]: I assume this is just the default location here (when coming directly from a link)?

File: src/routes/note-create.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      location,
+    } = this.props;
+
+    return location.state
+      ? (
+        <NoteCreatePage
+          referredEntityData={this.getReferredEntityData()}
+          entityTypesTranslationKeys={entityTypesTranslationKeys}
+          paneHeaderAppIcon="eholdings"
+          domain="eholdings"
```
Comment (carolegodfrey) [inline src/routes/note-create.js]: Wondering if domain (eholdings) will be used elsewhere and could be defined as a constant

File: test/bigtest/tests/note-create-test.js (hunk 3 above)
Comment (MaksymDryha) [reply test/bigtest/tests/note-create-test.js]: @carolegodfrey, yes, correct

File: src/routes/note-create.js (hunk 4 above)
Comment (MaksymDryha) [reply src/routes/note-create.js]: Nice suggestion! I've created constants which contain the names of the domain and app icon  associated with eholdings.

**Pull request merged**